### ISSUE TITLE
Should we test colour contrast of selected text?

### ISSUE THREAD

**Issue by Jym77** [act-rules/act-rules.github.io#1676]
Our current colour contrast rules check for contrast of text "as is". Foreground and background colours often change when text is selected and it is easy to make it with insufficient contrast.

The rules do not mention selected text at all. I implicitly assume they only look at unselected text (and this is what our implementation looks at).

It should be fairly easy to make a rule targeting **selected** text nodes (using the `::selection` pseudo-element, the same way we have [focused](https://act-rules.github.io/glossary/#focused) defined in term of the `:focus` pseudo-class). The Applicability being clear about "selected text" should also be clear that the test procedure requires selecting the text first.
It should also be doable to alter the existing rules (e.g. with a second expectation) to test the text node both when unselected and when selected. Not sure which version would be best.
It might however be difficult to build examples (since we want our examples to be testable "out of the box", so we need a way to programmatically select some text at load time which I'm not sure is doable…)

I'm also not fully sure that WCAG requires good contrast for selected text. It doesn't explicitly mention it…

**Comment by WilcoFiers** [comment]
I'm not so sure this would actually fail WCAG. Kinda doubt it to be honest. It seems a little strict.

**Comment by WilcoFiers** [comment]
Going to close this. Carlos and I both think this is not a failure of WCAG.
